core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -133,7 +133,7 @@ private[spark] class CoarseMesosSchedulerBackend(
     val driver = createSchedulerDriver(
       master,
       CoarseMesosSchedulerBackend.this,
-      sc.sparkUser,
+      conf.get("spark.executor.start.user", sc.sparkUser),
```

Comment:
I don't think we should introduce a config here. Whether it actually fixes your issue this is the wrong place to do it.